portfolio/src/main/client/src/components/sections/comments.ts

```diff
@@ -299,9 +300,38 @@ const Comment = (comment: CommentData): HTMLElement => {
     });
   }
 
-
   const readmore = ReadMore(new Text(comment.content));
-  readmore.root.classList.add('comment-content');
+
+  const btnsEl = htmlElement`
+    <div class="comment-controls">
+      ${readmore.root}
+    </div>`;
+
+  if ('contentLang' in comment && comment.contentLang in languages) {
+    const langInfo = languages[comment.contentLang as keyof typeof languages];
```

Comment:
is this `as keyof typeof languages` necessary? Seems like you already make this assert in your earlier if statement, so I'd expect typescript to make the type inference.

Reply:
Nope! Type narrowing only seems to work on variables, not properties of variables. It would work if I did:
```typescript
if ('contentLang' in comment) {
  const {contentLang} = comment;
  if (contentLang in languages) {
    // ...
  }
}
```

But that's even more verbose.